sips/pending/_posts/2017-01-11-refer-other-arguments-in-args.md

```diff
@@ -0,0 +1,100 @@
+---
+layout: sip
+disqus: true
+title: SIP-NN - Allow referring to other arguments in default parameters
+---
+
+**By: Pathikrit Bhowmick**
+
+## History
+
+| Date          | Version          |
+|---------------|------------------|
+| Jan 11th 2017 | Initial Draft    |
+| Jan 12th 2017 | Initial Feedback |
+
+## Introduction
+Currently there is no way to refer to other arguments in the default parameters list:
+
+Does not compile:
+```scala
+def substring(s: String, start: Int = 0, end: Int = s.length): String
+```
+
+The workaround to achieve this is by using a curried-function:
+```scala
+def substring(s: String, start: Int = 0)(end: Int = s.length): String
+```
+
+However, the above workaround is not always suitable in certain situations.
+
+The other more verbose alternative is by overloading:
+
+```scala
+def substring(s: String, start: Int = 0, end: Int = s.length): String
```

Comment:
This is not working for me, I get `<console>:12: error: not found: value s` in the console.